src/arch/x86/kvm/x86_cpu.cc

```diff
@@ -869,6 +876,22 @@ X86KvmCPU::updateKvmStateSRegs()
     setSpecialRegisters(sregs);
 }
 
+
+void
+X86KvmCPU::updateKvmStateAVXCommon(ThreadContext *tc, const kvm_xsave &kxsave)
+{
+    // Like updateKvmStateFPUCommon, updateKvmStateAVXCommon is called by
+    // updateKvmStateFPU function
+    // Setting Kvm's ymm reg values to gem5's ymm reg values
+    const AVXState *avx = (const AVXState*)((const char*)kxsave.region
+                            + sizeof(FXSave) + sizeof(XSaveHeader));
```

Comment:
This code makes me very uncomfortable. At a minimum, you should use `uint8_t` instead of `char`. However, I don't like doing this kind of pointer arithmetic at all. It's error prone and hard to debug. If it's at all possible to use a struct to get the offset to the AVXState it would be much better.

Reply:
I think `kxsave.region[sizeof(FXSave) + sizeof(XSaveHeader)]` would resolve this